util-images/request-benchmark/main.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+	"time"
+)
+
+const (
+	MonitorSleep  = 5 * time.Second
+	NamespaceTmpl = "%namespace%"
+)
+
+var (
+	inflight = flag.Int("inflight", 0, "Benchmark inflight (number of parallel requests being made to the apiserver")
+	uri      = flag.String("uri", "", "Request URI")
+
+	namespace = flag.String("namespace", "default", "Replace %namespace% in URI with provided namespace")
+
+	debug   = flag.Bool("debug", false, "Run in debug mode (more logging)")
+	monitor = flag.Bool("monitor", false, fmt.Sprintf("Monitor %s configmap for config changes", BenchmarkConfigMap))
+
+	responses = make(chan ResponseData)
+)
+
+type ResponseData struct {
+	Bytes    int
+	Duration time.Duration
+	Error    error
+}
+
+func main() {
+	flag.Parse()
+
+	client, err := GetClient()
+	if err != nil {
+		panic(err)
+	}
+
+	if *debug {
+		log.Println("Using client with config:", client.ConfigStr())
+	}
+
+	if err = client.CheckConnection(); err != nil {
+		panic(err)
+	}
+
+	if strings.Contains(*uri, NamespaceTmpl) {
+		log.Printf("Replacing namespace in %s with %s", *uri, *namespace)
+		newUri := strings.ReplaceAll(*uri, NamespaceTmpl, *namespace)
+		uri = &newUri
+	}
+	log.Printf("Sending requests to '%s' with inflight %d. Press Ctrl+C to stop...", *uri, *inflight)
+
+	for i := 0; i < *inflight; i++ {
```

Comment:
I think we should try to avoid having two separate actuations logic for:
* flag-initialized change
* configmap-initialized changes

IMHO we should either:
* wrap flags to the BenchmarkConfig struct and rely on monitorConfig
* get rid of flags interface at all and use configmaps as an input only?